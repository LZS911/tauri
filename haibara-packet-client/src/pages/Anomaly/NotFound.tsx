import { Button } from 'antd';
import { useTranslation } from 'react-i18next';
import { useNavigate } from 'react-router-dom';

const targetPath = '/dashboard';
const NotFound: React.FC = () => {
  const { t } = useTranslation();
  const navigate = useNavigate();
  const backToHome = () => {
    navigate(targetPath);
  };
  return (
    <div className="flex min-h-full flex-col items-center justify-center">
      <img
        title="not-found"
        src="/static/images/not-found.png"
        className="my-20"
      />

      <p className="text-4xl dark:text-white">{t('anomaly.notFound.title')}</p>

      <p className="my-4 w-[30%] text-center text-sm opacity-70 dark:text-white">
        {t('anomaly.notFound.subTitle')}
      </p>

      <Button type="primary" onClick={backToHome}>
        {t('anomaly.notFound.backToHome')}
      </Button>
    </div>
  );
};

export default NotFound;
