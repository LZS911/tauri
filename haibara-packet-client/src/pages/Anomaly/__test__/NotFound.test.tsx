import { fireEvent, screen } from '@testing-library/react';
import { useNavigate } from 'react-router-dom';
import { renderWithRouter } from '../../../testUtils/customRender';
import NotFound from '../NotFound';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: jest.fn(),
}));

describe('test NotFound', () => {
  const useNavigateMock: jest.Mock = useNavigate as jest.Mock;
  const navigateMock = jest.fn();

  beforeEach(() => {
    useNavigateMock.mockReturnValue(navigateMock);
  });

  afterEach(() => {
    jest.clearAllMocks();
    jest.clearAllTimers();
  });

  test('should match snapshot', () => {
    const { container } = renderWithRouter(<NotFound />);
    expect(container).toMatchSnapshot();
  });

  test('should jump to targetPath when clicking backHome button', async () => {
    renderWithRouter(<NotFound />);
    expect(navigateMock).toBeCalledTimes(0);
    fireEvent.click(screen.getByText('anomaly.notFound.backToHome'));
    expect(navigateMock).toBeCalledTimes(1);
    expect(navigateMock).toBeCalledWith('/dashboard');
  });
});
