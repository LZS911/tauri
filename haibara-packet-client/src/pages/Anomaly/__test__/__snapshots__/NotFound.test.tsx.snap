// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test NotFound should match snapshot 1`] = `
<div>
  <div
    class="flex min-h-full flex-col items-center justify-center"
  >
    <img
      class="my-20"
      src="/static/images/not-found.png"
    />
    <p
      class="text-4xl dark:text-white"
    >
      anomaly.notFound.title
    </p>
    <p
      class="my-4 w-[30%] text-center text-sm opacity-70 dark:text-white"
    >
      anomaly.notFound.subTitle
    </p>
    <button
      class="ant-btn ant-btn-primary theme-button-primary theme-button"
      type="button"
    >
      <span>
        anomaly.notFound.backToHome
      </span>
    </button>
  </div>
</div>
`;
